src/DistributedComputations/distributed_fft_based_poisson_solver.jl

```diff
@@ -1,28 +1,24 @@
-import PencilFFTs
-using PencilArrays: Permutation
-
 import FFTW 
 
+using CUDA: @allowscalar
+using Oceananigans.Grids: YZRegularRG
+
 import Oceananigans.Solvers: poisson_eigenvalues, solve!
 import Oceananigans.Architectures: architecture
+import Oceananigans.Fields: interior
 
-struct DistributedFFTBasedPoissonSolver{P, F, L, λ, S, I}
+struct DistributedFFTBasedPoissonSolver{P, F, L, λ, B, S}
     plan :: P
     global_grid :: F
     local_grid :: L
     eigenvalues :: λ
+    buffer :: B
     storage :: S
-    input_permutation :: I
```

Comment:
Why don't you need this anymore? PencilFFTs argue this is crucial for performance, but is that not true?

Reply:
I am not sure how PencilFFT works, but the largest cost in the distributed FFT is the MPI pass by far (the `Alltoallv` operation). All the rest is pretty fast in comparison, both the packing, the unpacking, and the transforms. I can take a look at what PencilFFT does but if that input permutation does not end up reducing the number of transposes it will not impact performance (at least for us).

Reply:
No that is a fine explanation.